src/constructors/regular.c

```diff
@@ -410,7 +411,7 @@ igraph_error_t igraph_square_lattice(
 #define IS_PERIODIC(dim) ((periodic && VECTOR(*periodic)[dim]))
 
     for (i = 0; i < no_of_nodes; i++) {
-        IGRAPH_ALLOW_INTERRUPTION();
+        IGRAPH_ALLOW_INTERRUPTION_LIMITED(iter, 1 << 10);
```

Comment:
Can't we use `i` here directly?

Reply:
If we do that, we need to use `i % skips == 0`. As I recall, `%` was slow, or perhaps not as amenable to compiler optimization across the loop. But I did not test this now.

Reply:
Good point!